CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required (VERSION 2.8)
+cmake_minimum_required (VERSION 2.8.12)
```

Comment:
that'll mean we'll only be linking them in the shared version again, right?

i think that's ok for now. having the find_package(Threads..) is an improvement even in the static case where we don't use them as not linking against it in your final executable causes runtime errors.